### PR TITLE
changed transient keys

### PR DESCRIPTION
**Pull request by volkanakb**
added wpml compatiblity
solved multilanguage issues
refactored parts of code

## TLDR; 

* transients weren't cleared for all languages on save

### CODE REVIEW

**Comment by szepeviktor** [comment]
> `if (defined('ICL_LANGUAGE_CODE')) {`

WPML is a no go for internationalization.

Will we support [Polylang](https://wordpress.org/plugins/polylang/)?

**Comment by d4ny0** [comment]
as @martinWelte once told we're an agency where we use this plugin mostly. most of the time we use wpml for internationalization but this might change in the future.

**Comment by szepeviktor** [comment]
Thank you.

Although there are many thousands of WPML users it is not a livable product.
The manufacturer (OnTheGoSystems) operates in an unsustainable manner.

**Comment by volkanakb** [comment]
added polylang compatibility, please check again
@szepeviktor @d4ny0

**Comment by szepeviktor** [comment]
Thank you.

File: src/Plugin.php [hunk 1]
```diff
@@ -237,13 +244,21 @@ public function saveOptionsHook(): void
     {
         $screen = \get_current_screen();
         if (false !== strpos($screen->id, 'towa-gdpr-plugin')) {
-            if (!isset($_POST['acf']['towa_gdpr_settings_hash']) || '' === $_POST['acf']['towa_gdpr_settings_hash'] || isset($_POST['save_and_hash'])) {
+            if (
```
Comment (SahinU88) [inline src/Plugin.php]: would extract this to a helper function so it becomes more readable.

File: src/Plugin.php [hunk 2]
```diff
@@ -256,14 +271,65 @@ public function saveOptionsHook(): void
         (new SettingsTableAdapter())->save();
     }
 
+    /**
+     * @param string|null $languageCode
+     * @return string
+     */
+    public static function getTransientKey(string $languageCode = null): string
+    {
+        if ($languageCode) {
```
Comment (SahinU88) [inline src/Plugin.php:273]: would extract this to a helper function or own class for the language. in there we can cover the logic to determine which plugin is used or use the fallback.

**Review by SahinU88 — CHANGES_REQUESTED** [review]
In general, it seems like we are using a lot of if-else statements. we should consider addding issues for refactoring and creating classes or thinking a bit about the structure/architecture for maintainability.

also maybe we could include the Str-helper library from laravel. saw a bit of code where wee could use it.

the 2 requests could be the beginning.

File: src/Plugin.php (hunk 1 above)
Comment (volkanakb) [reply src/Plugin.php]: done, please recheck

Comment (SahinU88) [reply src/Plugin.php]: @volkanakb where? 😅

File: src/Plugin.php [hunk 3]
```diff
@@ -235,18 +243,18 @@ private function loadTextdomain(): void
      */
     public function saveOptionsHook(): void
     {
-        $screen = \get_current_screen();
-        if (false !== strpos($screen->id, 'towa-gdpr-plugin')) {
-            if (!isset($_POST['acf']['towa_gdpr_settings_hash']) || '' === $_POST['acf']['towa_gdpr_settings_hash'] || isset($_POST['save_and_hash'])) {
+        if (PluginHelper::isGdprPluginAdminScreen()) {
```
Comment (volkanakb) [inline src/Plugin.php:246]: @SahinU88 here

**Review by SahinU88 — APPROVED** [review]
Looks good so far 👍 
I would probably start also doing some PRs for code-improvements along the way in general.

**Review by szepeviktor — APPROVED** [review]

**Pull request merged**